core-rust/state-manager/src/mempool/priority_mempool.rs

```diff
@@ -97,12 +98,26 @@ pub struct MempoolData {
     pub source: MempoolAddSource,
 }
 
+// There is only 1 [`MempoolData`] per unique [`MempoolTransaction`]
+impl Hash for MempoolData {
```

Comment:
This feels weird / scary to me that we have an `Eq` and `Hash` which disagree on equality checks.

I think the issue is that it seems wrong to me that `MempoolData` is a key in a hashmap.